kaiko/src/adapter.ts

```diff
@@ -46,19 +50,25 @@ const execute = (input, callback) => {
     'X-Api-Key': util.getRandomRequiredEnv('API_KEY'),
     'User-Agent': 'Chainlink',
   }
-  const config = {
+  const requestConfig = {
     url,
     params,
     headers,
     timeout: 10000,
   }
-  Requester.request(config, customError)
-    .then((response) => {
-      const result = response.data.data.filter((x) => x.price !== null)
-      response.data.result = Number(Requester.validateResultNumber(result, [0, 'price']))
-      callback(response.status, Requester.success(jobRunID, response))
-    })
-    .catch((error) => callback(500, Requester.errored(jobRunID, error)))
+
+  const response = await Requester.request(requestConfig, customError)
+  const result = (response.data.result = Number(
+    Requester.validateResultNumber(response.data.data, [0, 'price']),
+  ))
+  response.data.result = result
+  return Requester.success(jobRunID, {
+    data: response.data,
+    result,
+    status: 200,
+  })
```

Comment:
`validateResultNumber` already brings a `number`
```suggestion
  const result = Requester.validateResultNumber(response.data.data, [0, 'price'])
  return Requester.success(jobRunID, {
    data: {
       ...response.data,
       result,
    },
    result,
    status: 200,
  })
```